packages/shorebird_cli/lib/src/adb.dart

```diff
@@ -28,15 +28,32 @@ class Adb {
   }
 
   /// Starts the app with the given [package] name.
-  Future<void> startApp(String package) async {
-    final result = await _exec('shell monkey -p $package 1');
+  Future<void> startApp({
+    required String package,
+    String? deviceId,
+  }) async {
+    final args = [
+      if (deviceId != null) '-s $deviceId',
```

Comment:
This (combining two args into one string) will only work if shell=true I think.  Otherwise the args will get passed as separate strings and fail to parse I think?

Reply:
But otherwise this is a fantastic use of collection control flow statements. :).  I suspect the runInShell:false safe version is if (deviceId != null) ...['-s', deviceId]?  But then again, maybe I'm misremembering.

Reply:
I don't think this will matter in this particular case because we join all the args at the end before calling `exec`.

Reply:
Done